core/space/services/services.go

```diff
@@ -68,3 +70,44 @@ func NewSpace(
 		hub:      h,
 	}
 }
+
+var textileClientInitTimeout = time.Second * 60
+var textileClientHubTimeout = time.Second * 60 * 3
+
+// Waits for textile client to be initialized before returning.
+func (s *Space) waitForTextileInit(ctx context.Context) error {
+	if s.tc.IsInitialized() {
+		return nil
+	}
+
+	select {
+	case <-time.After(textileClientInitTimeout):
+		return errors.New("textile client not initialized in expected time")
+	case <-s.tc.WaitForInitialized():
```

Comment:
@dmerrill6 

Should we `s.tc.isInitialized = true` here?

Then, the `tc.onInitialized` chan would encapsulate the internal state.

Reply:
We are already doing that internally inside the `initialize` method of the Textile client, unless I'm misunderstanding what you are trying to say

Reply:
@dmerrill6 

You are right.